### PR TITLE
fix: Return None|False instead of UNAUTHORISED in the session recipe

### PR DESCRIPTION
**Pull request by KShivendu**
## Summary of change

Return `None|False` instead of `UNAUTHORISED` error in the session recipe

## Related issues

-   https://github.com/supertokens/backend/issues/83

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [x] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `supertokens_python/constants.py`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `setup.py`
    -   In `supertokens_python/constants.py`
-   [x] Had installed and ran the pre-commit hook
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If have added a new web framework, update the `supertokens_python/utils.py` file to include that in the `FRAMEWORKS` variable
-   [ ] If added a new recipe that has a User type with extra info, then be sure to change the User type in supertokens_python/types.py
 
## Remaining TODOs for this PR

-   [ ] Item1
-   [ ] Item2

### CODE REVIEW

**Comment by rishabhpoddar** [comment]
@KShivendu edit the PR comment to link to the issue and tick the checkboxes.

File: tests/test_session.py [hunk 1]
```diff
@@ -185,12 +185,14 @@ async def test_creating_many_sessions_for_one_user_and_looping():
 
     for handle in session_handles:
         info = await get_session_information(handle)
```
Comment (rishabhpoddar) [inline tests/test_session.py:187]: need to add more tests to make sure that update access token payload, update session data etc return the right boolean values based on the input.

File: supertokens_python/recipe/session/session_class.py [hunk 2]
```diff
@@ -33,6 +35,9 @@ async def get_session_data(
         session_info = await self.recipe_implementation.get_session_information(
             self.session_handle, user_context
         )
+        if session_info is None:
+            raise_unauthorised_exception("Session information does not exist.")
```
Comment (rishabhpoddar) [inline supertokens_python/recipe/session/session_class.py:39]: Error message should be `"Session does not exist anymore"` everywhere.

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: supertokens_python/recipe/session/session_class.py (hunk 2 above)
Comment (KShivendu) [reply supertokens_python/recipe/session/session_class.py]: done

File: tests/test_session.py (hunk 1 above)
Comment (KShivendu) [reply tests/test_session.py:175]: done.

**Pull request merged**